### PR TITLE
Add various Binaryen constant for new instructions

### PR DESCRIPTION
**Pull request by dcodeIO**
Also kind of a TODO list for instructions the Binaryen bindings don't yet support. Needing these for an IR experiment, so adding already.

- [x] I've read the contributing guidelines
- [x] I've added my name and email to the NOTICE file

### CODE REVIEW

File: src/module.ts [hunk 1]
```diff
@@ -1081,14 +1081,100 @@ export const enum RefAsOp {
...
+  /** string.as_wtf8 */
+  StringAsWTF8 = 0 /* TODO_BinaryenStringAsWTF8 */,
+  /** string.as_wtf16 */
+  StringAsWTF16 = 1 /* TODO_BinaryenStringAsWTF16 */,
+  /** string.as_iter */
+  StringAsIter = 2 /* TODO_BinaryenStringAsIter */
+}
+
+/** Binaryen StringIterMove operation constants. */
+export const enum StringIterMoveOp {
```
Comment (MaxGraey) [inline src/module.ts:1165]: How about just `StringIterOp`?

Comment (dcodeIO) [reply src/module.ts:1165]: So far these are 1:1 copies of what Binaryen has, where there is a `StringIterMove` instruction class taking these constants as their `op` field.

File: src/module.ts [hunk 2]
```diff
@@ -1081,14 +1081,100 @@ export const enum RefAsOp {
...
   /** br_on_data */
-  BrOnData = 3 /* TODO_BinaryenBrOnData */,
+  BrOnData = 6 /* TODO_BinaryenBrOnData */,
+  /** br_on_non_data */
+  BrOnNonData = 7 /* TODO_BinaryenBrOnNonData */,
   /** br_on_i31 */
-  BrOnI31 = 4 /* TODO_BinaryenBrOnI31 */
+  BrOnI31 = 8 /* TODO_BinaryenBrOnI31 */,
+  /** br_on_non_i32 */
+  BrOnNonI31 = 9 /* TODO_BinaryenBrOnNonI31 */
```
Comment (MaxGraey) [inline src/module.ts:1101]: Also wdyt refactor all sych ops from `BrOnOp.BrOnFunc` to `BrOnOp.Func` Due to `BrOnOp.BrOn...` and etc looks redundant

Comment (dcodeIO) [reply src/module.ts:1101]: Didn't do any renaming yet, but yeah, these renamings make sense to me :)

File: src/module.ts (hunk 1 above)
Comment (MaxGraey) [reply src/module.ts:1165]: ok

File: src/module.ts (hunk 2 above)
Comment (MaxGraey) [reply src/module.ts:1101]: Yeah. Better do this in separate PR

Comment (dcodeIO) [reply src/module.ts]: Did this here for the new ones not referenced otherwise now, but there's also `RefIsOp` and `RefAsOp` to update later.

**Pull request merged**